File: Scripts/server/factorial/perl/factorial.pl
foreach ( 0 .. 1000000000 ) {
   factorial( 10 );
}

sub factorial
{
   my $number = shift;   # get the argument

   if ( $number <= 1 ) { # base case
      return 1;
   }
   else {                # recursive step
      return $number * factorial( $number - 1 );
   }
}

File: Scripts/server/factorial/c++/factorial.cpp
#include <iostream>

int factorial(int);

int main() {
for (int i = 0; i < 1000000000; ++i) {	
	int j = factorial(10);
}
	return 0; 	
}

int factorial(int number) {
	int temp;

	if(number <= 1) return 1;

	temp = number * factorial(number - 1);
	return temp;
}

